launcher/ui/pages/modplatform/ModPage.cpp

```diff
@@ -241,6 +246,72 @@ void ModPage::onModSelected()
     ui->packView->adjustSize();
 }
 
+void ModPage::openUrl(const QUrl& url)
+{
+    // do not allow other url schemes for security reasons
+    if (!(url.scheme() == "http" || url.scheme() == "https")) {
+        qWarning() << "Unsupported scheme" << url.scheme();
+        return;
+    }
+
+    // detect mod URLs and search instead
+    static const QRegularExpression modrinth(QRegularExpression::anchoredPattern("(?:www\\.)?modrinth\\.com\\/mod\\/([^\\/]+)\\/?")),
+        curseForge(QRegularExpression::anchoredPattern("(?:www\\.)?curseforge\\.com\\/minecraft\\/mc-mods\\/([^\\/]+)\\/?")),
+        curseForgeOld(QRegularExpression::anchoredPattern("minecraft\\.curseforge\\.com\\/projects\\/([^\\/]+)\\/?"));
```

Comment:
```suggestion
    static const auto modrinth (QRegularExpression::anchoredPattern("(?:www\\.)?modrinth\\.com\\/mod\\/([^\\/]+)\\/?"));
    static const auto curseForge (QRegularExpression::anchoredPattern("(?:www\\.)?curseforge\\.com\\/minecraft\\/mc-mods\\/([^\\/]+)\\/?"));
    static const auto curseForgeOld (QRegularExpression::anchoredPattern("minecraft\\.curseforge\\.com\\/projects\\/([^\\/]+)\\/?"));
```

maybe you could also move those just outside the function (i.e. above line 249) to keep the function itself a bit tidier.

Reply:
Auto stops it from working correctly as it becomes a QString. You can still remove the comma declarations, but it becomes this
```c++
static const QRegularExpression modrinth(QRegularExpression::anchoredPattern("(?:www\\.)?modrinth\\.com\\/mod\\/([^\\/]+)\\/?"));
static const QRegularExpression curseForge(QRegularExpression::anchoredPattern("(?:www\\.)?curseforge\\.com\\/minecraft\\/mc-mods\\/([^\\/]+)\\/?"));
static const QRegularExpression curseForgeOld(QRegularExpression::anchoredPattern("minecraft\\.curseforge\\.com\\/projects\\/([^\\/]+)\\/?"));
```
I won't use spaces before the parentheses since clang-format changes them back.

Reply:
oh sorry, i thought it returned a QRegularExpression thingy :p

it was just to make the line a bit smaller, so its fine like that